README.md

```diff
@@ -118,3 +101,171 @@ Remote debugging using :2000
+
+For convenience, a cross-file for each supported hardware probe is available which provides a sane default configuration for it.
+
+The build configuration command for the native probe may look like:
+``` sh
+meson setup build --cross-file cross-file/native.ini
+```
+
+Alternatively (for advanced users), if you wish to configure manually, for instance while writing support for a new probe, or a different toolchain, you can run:
+``` sh
+meson setup build --cross-file cross-file/arm-none-eabi.ini -Dprobe=native
+```
+
+You now should have a `build` directory from where you can build the firmware, this is also where your binary files will appear.
+
+The command `meson compile` will build the default targets:
+```sh
+cd build
+meson compile
+```
+
+Ypu should now see the resulting binaries in `build`, in this case:
+```
+blackmagic_native_firmware.bin
+blackmagic_native_firmware.elf
+```
+
+These are the binary files you will use to flash to your probe.
+
+##### region `rom' overflowed
+
+It may happen, while working with non default configurations or the project latest version from git, that the firmware does not fit in the available space for the configured probe, this could look something like:
+```
+arm-none-eabi/bin/ld: region `rom' overflowed by 4088 bytes
+Memory region         Used Size  Region Size  %age Used
+             rom:      135160 B       128 KB    103.12%
+             ram:        5708 B        20 KB     27.87%
+collect2: error: ld returned 1 exit status
+```
+
+This is not unexpected, as some hardware probe have limited space for firmware, you can get around it by disabling some features or target support:
+``` sh
+meson setup --reconfigure -Dtargets=cortexm,stm -Drtt_support=false
+```
+
+#### Building Black Magic Debug App
+
+The Black Magic Debug App (BMDA) is always build by default, even for firmware builds, so long as all it's dependencies are found, you can find the executable under the build directory, named simply `blackmagic`
+
+If you wish to build only the BMDA, you can set the hardware `probe` option to an empty string `-Dprobe=''`, this is the default value:
+``` sh
+meson setup build
+```
+
+You now should have a `build` directory from where you can build the app, this is also where your executable will appear.
+
+The command `meson compile` will build the default targets:
+```sh
+cd build
+meson compile
+```
+
+Ypu should now see the resulting executable in `build`:
+```
+blackmagic
+```
+
+Most hosted building is done on and for Linux. BMP-hosted for windows can also be build with Mingw on Linux.
+
+### Changing the build configuration
+
+You may change the configuration at any time after configuring the project, if may also change the default options while configuring for the first time.
+
+Changing options at configure time:
+``` sh
+meson setup build -Ddebug_output=true
+```
+
+Changin options after configuration:
```

Comment:
`Changing`

Same comment for prefering `meson configure` to `meson setup --reconfigure`.

Reply:
Thanks, fixed the typo